src/components/Navigator/QuickNavigationModal.vue

```diff
@@ -9,180 +9,368 @@
-        class="close-icon"
-        @click="closeQuickNavigationModal()"
+        class="quick-navigation__container"
       >
-        <p>
-          Close
-        </p>
-      </div>
-      <input
-        ref="input"
-        class="filter"
-        v-model="userInput"
-        type="text"
-      />
-      <div>
+        <FilterInput
+          v-model="userInput"
+          class="quick-navigation__filter"
+          placeholder="Search symbols"
+          focusInputWhenCreated
+          focusInputWhenEmpty
+          @input="focusedIndex = 0"
+        >
+          <template #icon>
+            <div
+              class="quick-navigation__magnifier-icon-container"
+              :class="{ 'blue': userInput.length }"
+            >
+              <MagnifierIcon />
+            </div>
+          </template>
+          <template #input-menu-items>
+            <div
+              v-if="!userInput.length"
+              class="quick-navigation__open-key-container"
+            >
+              <kbd>Cmd+Shift+o</kbd>
+            </div>
+          </template>
+        </FilterInput>
         <div
-          v-for="(symbol, idx) in filteredSymbols"
-          class="symbol-match"
-          @click="closeQuickNavigationModal()"
-          :key="idx"
+          class="quick-navigation__match-list"
+          :class="{ 'active' : debouncedInput.length }"
         >
-          <Reference :url="symbol.path" :id="idx">
-            <span>
-              -
-              {{ symbol.title.slice(0, symbol.start) }}
-            </span>
-            <QuickNavigationHighlighter
-              :text="symbol.substring"
-              :matcherText="debouncedInput"
-            />
-            <span>
-              {{ symbol.title.slice(symbol.start + symbol.matchLength) }}
-            </span>
-          </Reference>
+          <div
+            v-if="noResultsWereFound"
+            class="no-results"
+          >
+            <p>
+              No results found.
+            </p>
+          </div>
+          <div
+            v-for="(symbol, index) in filteredSymbols"
+            :class="{ 'selected' : index == focusedIndex }"
+            :key="symbol.uid"
+            tabindex="0"
+            @click="closeQuickNavigationModal"
+            @focus.capture="focusIndex(index)"
+          >
+            <Reference
+              :url="symbol.path"
+              class="quick-navigation__reference"
+              tabindex="-1"
+            >
+              <div
+                class="quick-navigation__symbol-match"
+                ref="match"
+                role="list"
+              >
+                <div class="symbol-info">
+                  <div class="symbol-name">
+                    <NavigatorLeafIcon
+                      class="navigator-icon"
+                      :type="symbol.type"
+                    />
+                    <div class="symbol-title">
+                      <span v-text="symbol.title.slice(0, symbol.start)"></span>
```

Comment:
Calling native methods like this directly in the template can lead to some weird rendering issues sometimes, just wanted to point it out. Its a Vue thing, I have noticed over the years.